### ISSUE TITLE
Modern defaults when creating new angular application with strict flag

### ISSUE THREAD

**Issue by prabh-62** [angular/angular-cli#17589]
# 🚀 Feature request

### Command (mark with an `x`)

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description

When creating angular application with `strict` flag on, user still needs to update config files to strictest settings.

Here is a list of steps:

- `> npx @angular/cli@10.0.0-next.3 new ui-applications --createApplication=false --strict=true`
- `> cd ui-applications`
- `> yarn ng generate application admin-console --defaults=true`

#### Post actions after app is created

1. Edit `angular.json` and add `OnPush` as the default change detection strategy
	```json
	"@schematics/angular:component": {
	        "changeDetection": "OnPush"
	 }
	```

<img width="543" alt="1 angular-onpush" src="https://user-images.githubusercontent.com/13006107/80732831-db00fc80-8ada-11ea-8f91-d2062c1683b3.png">

2. Edit `angular.json` and update budgets 
   	```json
			  {
			   "type": "initial",
			   "maximumWarning": "500kb",
			   "maximumError": "600kb"
			  }	
	 ```
   <img width="416" alt="2 angular-budgets" src="https://user-images.githubusercontent.com/13006107/80733064-329f6800-8adb-11ea-8254-53d612b50ac7.png">

3) Edit `tsconfig.json` and set `strict` flag to true
    ```json
	      {
	        "compilerOptions": {
	          "strictNullChecks": true,
	          "strict": true
	        }
	      }
	 ```
   <img width="358" alt="3 tsconfig-strict" src="https://user-images.githubusercontent.com/13006107/80733117-4519a180-8adb-11ea-8ac3-fa62175fc509.png">

4. Edit `tsconfig.json` and set `strictTemplates` flag to true
      ```json
      "angularCompilerOptions": {
        "strictTemplates": true
       }
    ```
   <img width="342" alt="4 tsconfig-strictTemplates" src="https://user-images.githubusercontent.com/13006107/80733145-4b0f8280-8adb-11ea-8d18-fea41151b714.png">

5) Edit `tslint.json` and add accessibility rules from https://web.dev/accessible-angular-with-codelyzer/
     ```json
	    "template-accessibility-alt-text": true,
	    "template-accessibility-elements-content": true,
	    "template-accessibility-label-for": true,
	    "template-accessibility-tabindex-no-positive": true,
	    "template-accessibility-table-scope": true,
	    "template-accessibility-valid-aria": true,
	    "template-click-events-have-key-events": true,
	    "template-mouse-events-have-key-events": true,
	    "template-no-autofocus": true,
	    "template-no-distracting-elements": true
	```
   <img width="522" alt="5 codelyzer-a11y" src="https://user-images.githubusercontent.com/13006107/80733164-51056380-8adb-11ea-8e9c-0641e4e558b2.png">

6. Edit `projects/admin-console/.browserslistrc` and target modern browsers
     ```text
		  last 2 Edge versions
		  last 2 Chrome versions
		  last 2 Firefox versions
		  last 2 Safari versions
	  ```
   <img width="327" alt="6 browserlist" src="https://user-images.githubusercontent.com/13006107/80733175-5662ae00-8adb-11ea-96a5-fc852f2aefd6.png">

7) Edit `package.json` add lint, test and build scripts for `admin-console` project
  	 ```json
	    "test:admin-console": "ng test --project admin-console",
	    "lint:admin-console": "ng lint --project admin-console",
	    "build:admin-console": "ng build --prod --project admin-console"
	 ```
   <img width="571" alt="7 package-scripts" src="https://user-images.githubusercontent.com/13006107/80733185-5a8ecb80-8adb-11ea-8f54-02966fe69049.png">

I ran `yarn lint:admin-console` and `yarn build:admin-console` after making these changes(also, deleted contents of `app.component.html`) and I had no issues.
<img width="782" alt="9 yarn build_admin-console" src="https://user-images.githubusercontent.com/13006107/80734665-80b56b00-8add-11ea-8551-8ba94f5a0169.png">

### Describe the solution you'd like
If angular-cli could expose a `modern` (or something alike) flag, that would save developer time. It becomes very tedious to introduce these flags at a later stage when the team of developers is actively working on projects. 

- `> npx @angular/cli@10.0.0-next.3 new ui-applications --createApplication=false --strict=true --modern=true`

### Describe alternatives you've considered

I manually update config files for now

**Comment by alan-agius4** [comment]
Hi @prabh-62, 

Thanks for this request, great detail and explanation👍. You'll be excited to know that for version 10, we are doing some changes that address some of the above, we already have 2 PR in flight.

See the below PR for more context:
- https://github.com/angular/angular-cli/pull/17361
- https://github.com/angular/angular-cli/pull/17372

Regarding some of your other points;

**Stricter Budgets**
Myself and @IgorMinar had a discussion around this week and indeed when using `strict`, we'll be setting stricter budgets.

**OnPush as default change detection strategy**
I think this is a good idea which we didn't discuss yet, @IgorMinar what are your thoughts?

**Adding Codelyzer accessibility rules**
I quite dig this, thoughts @mgechev and @IgorMinar?

**Edit package.json add lint, test and build scripts for admin-console project**
I don't see any added value with the new scripts in the package.json. As you can already achieve the same thing if you run:

```cmd
yarn lint admin-console
```
or
```cmd
ng lint admin-console
```

**Comment by mgechev** [comment]
I'd suggest holding off on stricter codelyzer/tslint. Both tools' maintenance would be discontinued. We should consider linting after we migrate to eslint.

**Comment by alan-agius4** [comment]
Hi @prabh-62, 

Most of the desired changes has been implemented in the following 2 PRs #17361 and #17387. 

We have discussed the remaining points with the team yesterday and we feel that we should hold off from adding them for following reasons:

**Stricter Codelyzer/Tslint rules** 
Both of tools will be discontinued, and we should consider linting after we migrate to ESLint. 

**On Push changeDetection**
While, indeed this is recommended in many cases from a performance standpoint, having this by default will make it harder for new users to adopt Angular because you'd need to take care of trigger change detection. In addition to that, this change will invalid a lot of courses, documentation and other learning material so this change alone might have a big impact on the community.

For these reason, I think we should close this issue for the time being and return to it in the future.
